### PR TITLE
CASMINST-6699: Ensure test RPMs are updated on masters

### PR DESCRIPTION
**Pull request by mharding-hpe**
Backport of https://github.com/Cray-HPE/docs-csm/pull/4371

### CODE REVIEW

File: upgrade/scripts/rebuild/ncn-rebuild-master-nodes.sh
Comment (rustydb) [inline upgrade/scripts/rebuild/ncn-rebuild-master-nodes.sh:1]: For 1.5 and 1.6, a node rebuild will install the latest version of the RPMs listed here: https://github.com/Cray-HPE/csm/blob/release/1.6/rpm/cloud-init.yaml#L28

Right now that list includes all of your RPMs except `craycli`, I'd suggest adding `craycli` to that list for 1.5 and 1.6 instead.

File: upgrade/scripts/rebuild/ncn-rebuild-master-nodes.sh [hunk 1]
```diff
@@ -184,6 +184,9 @@ else
   echo "====> ${state_name} has been completed"
 fi
 
+# Update test RPMs on the newly upgraded master
+update_test_rpms $target_ncn
```
Comment (rustydb) [inline upgrade/scripts/rebuild/ncn-rebuild-master-nodes.sh:188]: Same comment as `ncn-rebuild-master-nodes.sh`

File: upgrade/scripts/common/ncn-common.sh [hunk 2]
```diff
@@ -87,128 +85,145 @@ export STABLE_NCN=$(hostname)
...
-      record_state "${state_name}" ${target_ncn}
+function update_test_rpms() {
+  local state_name state_recorded target_ncn
+  target_ncn=$1
+  state_name="UPDATE_TEST_RPMS"
+  state_recorded=$(is_state_recorded "${state_name}" ${target_ncn})
+  if [[ $state_recorded == "0" ]]; then
+    echo "====> ${state_name} ..."
+    {
+      ssh $target_ncn '/usr/share/doc/csm/upgrade/scripts/upgrade/util/upgrade-test-rpms.sh --local'
```
Comment (rustydb) [inline upgrade/scripts/common/ncn-common.sh:221]: If this is ran after the node rebuilds, then we don't need it here. Cloud-init will installs these, but we have to add craycli and hpe-csm-goss-package to https://github.com/Cray-HPE/csm/blob/release/1.6/rpm/cloud-init.yaml#L28

File: install/deploy_final_non-compute_node.md [hunk 3]
```diff
@@ -434,7 +435,17 @@ it is used for Cray installation and bootstrap.
...
-## 6. Configure DNS and NTP on each BMC
+## 6. Update select RPMs
+
+(`ncn-m001`) Run the following command to ensure that select RPMs on `ncn-m001` are at the correct version.
+
+```bash
+/usr/share/doc/csm/upgrade/scripts/upgrade/util/upgrade-test-rpms.sh --local
+```
+
+If the output ends with `PASSED`, then it was successful, despite any warning messages that may have been displayed.
```
Comment (rustydb) [inline install/deploy_final_non-compute_node.md:446]: m001 will grab the latest RPMs based on cloud-init, all that's missing in cloud-init right now is `craycli`. I think all you have to do is add craycli to https://github.com/Cray-HPE/csm/blob/release/1.6/rpm/cloud-init.yaml

**Review by rustydb — CHANGES_REQUESTED** [review]

File: install/deploy_final_non-compute_node.md (hunk 3 above)
Comment (mharding-hpe) [reply install/deploy_final_non-compute_node.md:446]: https://github.com/Cray-HPE/csm/pull/2973

**Comment by mharding-hpe** [comment]
Addressing this per Rusty's comments with https://github.com/Cray-HPE/csm/pull/2973

**Pull request closed without merging**